### PR TITLE
8258134: assert(size == calc_size) failed: incorrect size calculation on x86_32 with AVX512 machines

### PR DESCRIPTION
**Pull request by DamonFool**
Hi all,

Two vector api tests crashed on x86_32 with AVX512 machines due to this assert [1].
The reason is that 'calc_size' is incorrect.

But there is no need to calculate 'calc_size' manually at all since the result [2] is actually never used by the VM.
Also, it is really hard to maintain the calculation logic for various hardwares and configurations.
And it may be easily broken again in the future with more and more complicated instructions & configurations.

So it would be better to remove the calculation and the assert, which is safe and already done for x86_64 [3].
The fix follows what is done for x86_64.
And it also makes vec_stack_to_stack_helper, vec_mov_helper and vec_spill_helper return void according to @neliasso 's comments [4].

Thanks.
Best regards,
Jie

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/x86_32.ad#L1016
[2] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/x86_32.ad#L1059
[3] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/x86_64.ad#L1042
[4] https://github.com/openjdk/jdk/pull/1753#discussion_r542505836

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8258134](https://bugs.openjdk.java.net/browse/JDK-8258134): assert(size == calc_size) failed: incorrect size calculation on x86_32 with AVX512 machines

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk16 pull/21/head:pull/21`
`$ git checkout pull/21`

### CODE REVIEW

**Comment by DamonFool** [comment]
Testing:
  - tier1~tier3 on Linux/x86_{32, 64}, no regression

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by DamonFool** [comment]
Thanks @vnkozlov for your review.

@neliasso , are you also OK with this change?
Thanks.

**Comment by DamonFool** [comment]
May I get a second review for this fix?
Thanks.

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Comment by DamonFool** [comment]
Thanks @TobiHartmann for your review.
/integrate

**Comment by kevinrushforth** [comment]
This breaks the build for JDK 16 on Windows.

I see that the pre-submit tests were not run on this pull request. See the following information comment from the Skara bot in the [checks](https://github.com/openjdk/jdk16/pull/21/checks?check_run_id=1567324083) section for more information.

**Comment by TobiHartmann** [comment]
> This breaks the build for JDK 16 on Windows.
> 
> I see that the pre-submit tests were not run on this pull request. See the following information comment from the Skara bot in the [checks](https://github.com/openjdk/jdk16/pull/21/checks?check_run_id=1567324083) section for more information.

I've filed [JDK-8258688](https://bugs.openjdk.java.net/browse/JDK-8258688).

**Comment by TobiHartmann** [comment]
Okay, [JDK-8258687](https://bugs.openjdk.java.net/browse/JDK-8258687) has already been filed for this.

**Comment by DamonFool** [comment]
Okay, I'll have a look. Sorry for this bug.

**Comment by TobiHartmann** [comment]
The problem is that `extern int vec_spill_helper` has still `int` in the signature.

**Comment by DamonFool** [comment]
> The problem is that `extern int vec_spill_helper` has still `int` in the signature.

Thanks @TobiHartmann for your help.
https://github.com/openjdk/jdk16/pull/51

I didn't know why the build tests were skipped for both pull/21 and pull/51.

**Comment by kevinrushforth** [comment]
> I didn't know why the build tests were skipped for both pull/21 and pull/51.

You need to enable the GitHub actions to run on your repo. If you go to the checks link I added above, you will see:

_In order to run pre-submit tests, the source repository must be properly configured to allow test execution. See https://wiki.openjdk.java.net/display/SKARA/Testing for more information on how to configure this._

**Comment by DamonFool** [comment]
> > I didn't know why the build tests were skipped for both pull/21 and pull/51.
> 
> You need to enable the GitHub actions to run on your repo. If you go to the checks link I added above, you will see:
> 
> _In order to run pre-submit tests, the source repository must be properly configured to allow test execution. See https://wiki.openjdk.java.net/display/SKARA/Testing for more information on how to configure this._

OK.
Thanks for your guidance.
I'll fix it next time.

**Pull request closed without merging**
